internal/probes/mcptransport/originvalidation.go

```
@@ -0,0 +1,663 @@
+		// context-deadline error AFTER the response headers arrived is not a
+		// probe failure — record what we have and let the classifier decide.
+		if transport == "sse" && resp != nil {
+			// Fall through to classify below.
+		} else {
+			a.SetError(fmt.Errorf("%s %s: %w", method, endpoint, err))
+			return a
+		}
+	}
+	defer func() {
+		if resp != nil {
+			resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)